### ISSUE TITLE
🐛 Internal panic on Astro's prop spread syntax

### ISSUE THREAD

**Issue by Frixuu** [biomejs/biome#7851]
### Environment information

```block
CLI:
  Version:                      2.3.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         alacritty
  JS_RUNTIME_VERSION:           v24.9.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.18.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

Trying to `lint` the following .astro file [(playground link)](https://biomejs.dev/playground/?code=LQAtAC0ACgBjAG8AbgBzAHQAIAB7ACAALgAuAC4AcgBlAHMAdAAgAH0AIAA9ACAAQQBzAHQAcgBvAC4AcAByAG8AcABzADsACgAtAC0ALQAKAAoAPABkAGkAdgAgAHsALgAuAC4AcgBlAHMAdAB9AD4APAAvAGQAaQB2AD4ACgA%3D&language=astro):

```astro
---
const { ...rest } = Astro.props;
---

<div {...rest}></div>
```

fails with the following message:

```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /__w/biome/biome/crates/biome_parser/src/lib.rs:571:9
Thread Name: biome::workspace_worker_0
Message: The parser is no longer progressing. Stuck at '{' L_CURLY:47..48

src/components/Repro.astro internalError/panic  INTERNAL
```

<details>
<summary>(Contents of biome.json, just in case)</summary>

```json
{
  "$schema": "https://biomejs.dev/schemas/2.3.0/schema.json",
  "html": {
    "experimentalFullSupportEnabled": true
  },
  "files": {
    "includes": ["**", "!!**/dist"]
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true
    }
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "double"
    }
  }
}
```
</details>

### Expected result

I expected Biome not to print that error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
The parsing for this was made available in #7872, but its not currently enabled for astro files.

**Comment by ematipico** [comment]
Closing. The issues has been fixed in one of latest versions of Biome.
